From 892eea28f600adf12b1b120e968834637cb17ecb Mon Sep 17 00:00:00 2001
From: Viorel Suman <viorel.suman@nxp.com>
Date: Mon, 29 Jan 2018 12:35:09 +0200
Subject: [PATCH 3264/5242] MLK-17445: ASoC: ak4497: let codec start/stop
 properly

commit  3952be6686c602a4f220ef627781b503770929b8 from
https://source.codeaurora.org/external/imx/linux-imx.git

Let the codec start/stop properly before syncing
regmap with codec registers.

Signed-off-by: Viorel Suman <viorel.suman@nxp.com>
Reviewed-by: Shengjiu Wang <shengjiu.wang@nxp.com>
Signed-off-by: Meng Li <Meng.Li@windriver.com>
---
 sound/soc/codecs/ak4497.c |    5 ++++-
 1 file changed, 4 insertions(+), 1 deletion(-)

diff --git a/sound/soc/codecs/ak4497.c b/sound/soc/codecs/ak4497.c
index 7bbde80..74145a6 100644
--- a/sound/soc/codecs/ak4497.c
+++ b/sound/soc/codecs/ak4497.c
@@ -867,6 +867,7 @@ static int ak4497_runtime_suspend(struct device *dev)
 
 	if (ak4497->pdn_gpio > 0) {
 		gpio_set_value(ak4497->pdn_gpio, 0);
+		usleep_range(1000, 2000);
 	}
 
 	if (ak4497->mute_gpio > 0)
@@ -882,8 +883,10 @@ static int ak4497_runtime_resume(struct device *dev)
 	if (ak4497->mute_gpio > 0)
 		gpio_set_value(ak4497->mute_gpio, 1); /* External Mute ON */
 
-	if (ak4497->pdn_gpio > 0)
+	if (ak4497->pdn_gpio > 0) {
 		gpio_set_value(ak4497->pdn_gpio, 1);
+		usleep_range(1000, 2000);
+	}
 
 	regcache_cache_only(ak4497->regmap, false);
 	regcache_mark_dirty(ak4497->regmap);
-- 
1.7.9.5

